### PR TITLE
[doc] Add SCA guidelines for OTBN from COCO-Alma work.

### PR DESCRIPTION
**Pull request by jadephilipoom**
This text is from an OpenTitan-domain [doc](https://docs.google.com/document/d/1kX2fr5lH9xzYcpu2ycsac4JZ2j8L0Vl_0UNmoN3-UlY/edit) that @abdullahvarici wrote a while back, but it seems useful to have on the public documentation website as well.

### CODE REVIEW

**Review by nasahlpa — APPROVED** [review]
Thanks, makes sense to add this to the documentation!

File: doc/contributing/style_guides/otbn_style_guide.md [hunk 1]
```diff
@@ -146,6 +146,168 @@ Prefer `.word` over alternatives such as `.quad`.
...
+bn.xor w4, w0, w0
+bn.xor w5, w1, w1
+
+/* No leakage */
+bn.xor w4, w0, w0
+bn.xor w9, w9, w9 /* dummy instruction */
+bn.xor w5, w1, w1
+```
+
+**4. Do not have shares even in different bits of a register.**
```
Comment (rswarbrick) [inline doc/contributing/style_guides/otbn_style_guide.md:191]: Nit: I slightly struggled to understand this. Do we mean "Don't combine shares in a register, even in different bits" ?

**Review by rswarbrick — APPROVED** [review]
Eek, these look like hard guidelines to follow.

I wonder whether it would make sense to add some checking tool that could look at a trace and figure out whether the guidelines are all being followed. Practically speaking, we'd need to be able to tell it things like "w0 and w1 are two shares of a secret". But I don't think we'd need much more, so maybe this is a plausible tool to build?

File: doc/contributing/style_guides/otbn_style_guide.md (hunk 1 above)
Comment (jadephilipoom) [reply doc/contributing/style_guides/otbn_style_guide.md:191]: Yes, that's my understanding.

**Comment by jadephilipoom** [comment]
> I wonder whether it would make sense to add some checking tool that could look at a trace and figure out whether the guidelines are all being followed. Practically speaking, we'd need to be able to tell it things like "w0 and w1 are two shares of a secret". But I don't think we'd need much more, so maybe this is a plausible tool to build?

I agree! It would be good to be able to check these automatically. I think we'd need a bit more than just marking the initial share registers -- we'd have to track the information flow a little bit through the program to make sure no values that _flow from_ one share are combined with any that flow from the other share. As a short-term solution, we could indeed potentially program a quick and dirty Python checker to look for violations of these specific rules, since there's already information-flow logic and similar checks in Python for OTBN. Fundamentally it's not _too_ different of a problem than the constant-time checker we already have, it just needs to check two threads of "secret" instead of one and have different failure conditions.

Longer-term though, I think it would be really nice to have a formalized contract between OTBN hardware and software that specifies which values' interactions might be visible through side channels, so that hardware and software can both be tested and verified against the same concrete set of statements -- especially relevant as OTBN evolves, since the contract might change between versions. Curious what you think of that @rswarbrick ?

**Pull request merged**